### PR TITLE
test(@neo-one/client-core): Add coverage for @neo-one/client-core

### PR DESCRIPTION
**Pull request by danwbyrne**
### Description of the Change

adds unit test coverage for much of `@neo-one/client-core`. 

Want to see what codecov says about a few overlapping ones so submitting PR now.

### Benefits

coverage

### Applicable Issues

#566

### CODE REVIEW

File: packages/neo-one-client-core/src/__tests__/Client.test.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        network: 'test',
+      }),
+    }),
+  });
+
+  let providerOne: Modifiable<LocalUserAccountProvider<KeyStore, Provider>>;
+  let providerTwo: Modifiable<LocalUserAccountProvider<KeyStore, Provider>>;
+  let client: Client;
+  beforeEach(() => {
+    providerOne = new LocalUserAccountProvider(createStoreAndProvider(walletOne));
```
Comment (dicarlo2) [inline packages/neo-one-client-core/src/__tests__/Client.test.ts]: Generally you want to mock the inputs to the class/function under test. Change this to just mock `UserAccountProvider` directly. Look at the neo-one-client-full-core Client.test.ts file for an example.

File: packages/neo-one-client-core/src/errors.ts [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { makeErrorWithCode } from '@neo-one/utils';
 import BigNumber from 'bignumber.js';
 
 export const UnknownNetworkError = makeErrorWithCode('UNKNOWN_NETWORK', (name: string) => `Unknown network ${name}`);
+export const UnknownAddressError = makeErrorWithCode(
```
Comment (dicarlo2) [inline packages/neo-one-client-core/src/errors.ts]: We probably don't need both `UnknownAddressError` and `UnknownAccountError` - let's just stick with `UnknownAccountError`

File: packages/neo-one-client-core/src/__tests__/provider/JSONRPCHTTPProvider.test.ts [hunk 3]
```diff
@@ -15,9 +16,11 @@ describe('JSONRPCHTTPProvider', () => {
   };
 
   let provider = new JSONRPCHTTPProvider(endpoint);
+  let monitor: Monitor;
   // tslint:disable-next-line no-any
   const fetchMock: jest.Mock<typeof fetch> = fetch as any;
   beforeEach(() => {
+    monitor = DefaultMonitor.create({ service: 'test' });
```
[the remark this answers is not in the document]
Comment (danwbyrne) [reply packages/neo-one-client-core/src/__tests__/provider/JSONRPCHTTPProvider.test.ts]: I actually only use it for this one test, is it acceptable to just create it once in that test?

File: packages/neo-one-client-core/src/provider/NEOONEDataProvider.ts [hunk 4]
```diff
@@ -690,6 +693,7 @@ export class NEOONEDataProvider implements DeveloperProvider {
     return { transaction, verifyResult };
   }
 
+  /* istanbul ignore next */
```
[the remark this answers is not in the document]
Comment (danwbyrne) [reply packages/neo-one-client-core/src/provider/NEOONEDataProvider.ts]: added a test to cover it.

**Pull request merged**